AppDB/datastore_server.py

```diff
@@ -550,16 +522,16 @@ def get_table_prefix(self, data):
       A valid table prefix.
     """
     if isinstance(data, entity_pb.EntityProto):
-      data = data.key()
-
-    if not isinstance(data, tuple):
+      app_id = clean_app_id(data.key().app())
+      namespace = data.key().name_space()
+    elif isinstance(data, tuple):
+      app_id = data[0]
+      namespace = data[1]
+    else:
       app_id = clean_app_id(data.app())
-      data = (app_id, data.name_space())
-
-    prefix = "{0}{2}{1}".format(data[0], data[1], 
-      self._SEPARATOR).replace('"', '""')
+      namespace = data.name_space()
 
-    return prefix
+    return self._SEPARATOR.join([app_id, namespace]).replace('"', '""')
```

Comment:
I'll have to track down the reason for it myself. I was just keeping the same behavior.